support-lib/CMakeLists.txt

```diff
@@ -0,0 +1,76 @@
+cmake_minimum_required(VERSION 3.6.0)
+
+project(Djinni)
+
+include(GNUInstallDirs)
+
+set(SRC_SHARED
+  "djinni_common.hpp"
+  "proxy_cache_interface.hpp"
+  "proxy_cache_impl.hpp"
+)
+
+set(SRC_JNI
+  "jni/djinni_support.hpp"
+  "jni/Marshal.hpp"
+  "jni/djinni_support.cpp"
+)
+
+set(SRC_OBJC
+  "objc/DJICppWrapperCache+Private.h"
+  "objc/DJIError.h"
+  "objc/DJIMarshal+Private.h"
+  "objc/DJIObjcWrapperCache+Private.h"
+  "objc/DJIError.mm"
+  "objc/DJIProxyCaches.mm"
+)
+
+option(DJINNI_STATIC_LIB "Build Djinni as a static library instead of dynamic (the default)." off)
+if(DJINNI_STATIC_LIB)
+  add_library(djinni STATIC ${SRC_SHARED})
```

Comment:
I'm not a CMake expert, so I'm not sure what's the answer to this, but the projects that I use, tend to do configuration with options.